test/infrastructure/RepositoryFactory.spec.ts

```diff
@@ -70,6 +72,17 @@ describe('RepositoryFactory', () => {
         expect(repositoryFactory.createFinalizationRepository()).to.be.not.null;
     });
 
+    it('Raise error without unhandled-rejections', async () => {
+        const url = 'https://www.google.com';
+        const repositoryFactory = new RepositoryFactoryHttp(url);
+        try {
+            await repositoryFactory.createNodeRepository().getNodeHealth().toPromise();
+            expect(true).to.be.false;
```

Comment:
Should we use [chai-as-promised](https://www.chaijs.com/plugins/chai-as-promised/) or a similar promise assertion library as a better practice in general for promise assertions?

Reply:
Thanks! chai as promises looks nice. Although the try-catch could be more end-user-style assertion? And I wouldn't add another lib just yet. Let's consider it for the future.